File: cracking-code-interview/big-o/example-1.js
// The runtime will be O(n), the fact that it does run for twice won't matter, gets rid of constants
function sumAndProduct(array) {
  let sum = 0;
  let product = 1;
  for (let i in array) {
    sum += array[i];
  }
  for (let i in array) {
    product *= array[i];
  }
  console.log(`sum = ${sum}, product = ${product}`);
}

let array = [1, 5, 8, 9, 11, 13, 15, 19, 21];
sumAndProduct(array);

File: cracking-code-interview/big-o/example-1.rb
# The runtime will be O(n), the fact that it does run for twice won't matter, gets rid of constants
def sum_and_product(array)
  sum = 0
  product = 1

  array.each do |el|
    sum += el
  end
  array.each do |el|
    product *= el
  end
  puts("sum = #{sum}, product = #{product}")
  [sum, product]
end

array = [1, 5, 8, 9, 11, 13, 15, 19, 21]
sum_and_product(array)
